www/gatsby-node.js

```diff
@@ -615,18 +625,109 @@ exports.createPages = ({ graphql, actions, reporter }) => {
       })
 
       // Create docs pages.
-      result.data.allMdx.edges.forEach(edge => {
-        const slug = _.get(edge, `node.fields.slug`)
+      const docPages = result.data.allMdx.edges
+      const docLinks = docLinksData[0].items
+      const tutorialLinks = tutorialLinksData[0].items
+      const contributingLinks = contributingLinksData[0].items
+
+      // flatten sidebar links trees for easier next/prev link calculation
+      function flattenList(itemList) {
```

Comment:
I tried swapping out `flatten` but it only flattens one level deep, there is also `flattenDeep`, but it looks like you have to specify the depth to flatten, which would come back to a situation like I ran into trying to query for the yaml tree where we have to explicitly add in the depth. The custom flatten is more true recursion so I think it might be a better fit. 

I could certainly move this function elsewhere but wasn't sure it'd be super useful anywhere else, where do you think it would live?  🤷‍♂